src/components/ProgressiveImage/index.tsx

```diff
@@ -0,0 +1,62 @@
+import React, {useState} from 'react';
+import clsx from 'clsx';
+
+import './ProgressiveImage.scss';
+
+type Props = {
+  alt: string;
+  placeholderSrc: string; // src to a much smaller version of the real image
+  realSrc: string;
+  width?: number;
+  height?: number;
+  containerClassName?: string;
+  placeholderImageClassName?: string;
+  realImageClassName?: string;
+};
+
+const ProgressiveImage = ({
+  alt,
+  containerClassName,
+  height,
+  placeholderImageClassName,
+  placeholderSrc,
+  realImageClassName,
+  realSrc,
+  width,
+}: Props) => {
+  const [isLoaded, setIsLoaded] = useState(false);
+  return (
+    <div className={clsx('ProgressiveImage', containerClassName)} style={{height, width}}>
```

Comment:
Ideally it would be cool if we can use this `ProgressiveImage` component just like how we use `<img />`. Is there a reason why this component needs this outer `div`? Without it it can behave much more like a native `img`

Reply:
This component needs an outer `div` as there are two overlapping images, one for the placeholder and one for the real image. Both of them need to be `position: absolute` to overlap each other, which means that there needs to be a relatively positioned outer div.

Reply:
ah makes sense